### PR TITLE
Add an E2E test to cover volume resize

### PR DESCRIPTION
**Pull request by sebgl**
The test relies on one of the available storage class that supports
volume expansion, or is skipped if there's none.

Needs https://github.com/elastic/cloud-on-k8s/pull/3752 to be merged first.
Relates https://github.com/elastic/cloud-on-k8s/issues/325.

### CODE REVIEW

File: test/e2e/es/volume_test.go [hunk 1]
```diff
@@ -21,3 +31,115 @@ func TestVolumeEmptyDir(t *testing.T) {
...
+	// resize the volume to 1Gi after the cluster is up
+	initialStorageSize := b.Elasticsearch.Spec.NodeSets[0].VolumeClaimTemplates[0].Spec.Resources.Requests.Storage()
+	resizedStorage := initialStorageSize.DeepCopy()
+	resizedStorage.Add(resource.MustParse("1Gi"))
+
+	test.Sequence(nil, func(k *test.K8sClient) test.StepList {
+		return test.StepList{
+			{
+				Name: "Update the Elasticsearch spec with increased storage requests",
+				Test: test.Eventually(func() error {
```
Comment (barkbay) [inline test/e2e/es/volume_test.go:69]: Are we using `test.Eventually` to retry on update conflict ?

File: test/e2e/es/volume_test.go [hunk 2]
```diff
@@ -21,3 +31,115 @@ func TestVolumeEmptyDir(t *testing.T) {
...
+				Name: "PVCs should eventually be resized",
+				Test: test.Eventually(func() error {
+					for _, pvcName := range pvcNames {
+						var pvc corev1.PersistentVolumeClaim
+						if err := k.Client.Get(types.NamespacedName{Namespace: b.Elasticsearch.Namespace, Name: pvcName}, &pvc); err != nil {
+							return err
+						}
+						reportedStorage := pvc.Status.Capacity.Storage()
+						if reportedStorage == nil {
+							return fmt.Errorf("no ostorage size reported in %s status", pvcName)
```
Comment (barkbay) [inline test/e2e/es/volume_test.go:88]:
```suggestion
							return fmt.Errorf("no storage size reported in %s status", pvcName)
```

File: test/e2e/es/volume_test.go [hunk 3]
```diff
@@ -21,3 +31,115 @@ func TestVolumeEmptyDir(t *testing.T) {
...
+			{
+				Name: "StatefulSets should eventually be recreated with the right storage size",
+				Test: test.Eventually(func() error {
+					for _, ssetName := range []string{masterSset, dataSset} {
+						var sset appsv1.StatefulSet
+						if err := k.Client.Get(types.NamespacedName{Namespace: b.Elasticsearch.Namespace, Name: ssetName}, &sset); err != nil {
+							return err
+						}
+						if !sset.Spec.VolumeClaimTemplates[0].Spec.Resources.Requests.Storage().Equal(resizedStorage) {
+							return fmt.Errorf("StatefulSet %s has not been recreatd with storage size %s", ssetName, resizedStorage.String())
```
Comment (barkbay) [inline test/e2e/es/volume_test.go:106]:
```suggestion
							return fmt.Errorf("StatefulSet %s has not been recreated with storage size %s", ssetName, resizedStorage.String())
```

**Comment by barkbay** [comment]
Tested on OCP 4: ✅

File: test/e2e/es/volume_test.go (hunk 1 above)
Comment (sebgl) [reply test/e2e/es/volume_test.go:69]: Yes! Also to account for any network glitch.

**Comment by sebgl** [comment]
I'm testing it on Azure. I suspect it may not work since Azure does not support proper resizing even though their storage classes specify `allowVolumeExpansion: true`.

**Comment by sebgl** [comment]
It seems to work fine on Azure with the `azurefile` storage class picked up by default 👍

**Review by barkbay — APPROVED** [review]
LGTM

**Pull request merged**